starboard/nplb/player_write_sample_test.cc

```diff
@@ -31,95 +33,114 @@ typedef SbPlayerTestFixture::VideoSamples VideoSamples;
+  if (player_fixture.HasVideo()) {
+    samples_to_write = std::min(
+        samples_to_write, SbPlayerGetMaximumNumberOfSamplesPerWrite(
+                              player_fixture.GetPlayer(), kSbMediaTypeAudio));
+  }
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(
+      player_fixture.HasAudio() ? AudioSamples(0, samples_to_write).WithEOS()
+                                : AudioSamples(),
+      player_fixture.HasVideo() ? VideoSamples(0, samples_to_write).WithEOS()
+                                : VideoSamples()));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
 }
 
 TEST_P(SbPlayerWriteSampleTest, WriteMultipleBatches) {
-  if (test_media_type_ == kSbMediaTypeAudio) {
-    ASSERT_NO_FATAL_FAILURE(
-        player_fixture_->Write(AudioSamples(0, 8).WithEOS()));
-  } else {
-    SB_DCHECK(test_media_type_ == kSbMediaTypeVideo);
-    ASSERT_NO_FATAL_FAILURE(
-        player_fixture_->Write(VideoSamples(0, 8).WithEOS()));
+  const int kSamplesToWrite = 8;
+
+  SbPlayerTestFixture player_fixture(GetParam());
+  EXPECT_FALSE(HasFailure());
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(
+      player_fixture.HasAudio() ? AudioSamples(0, kSamplesToWrite).WithEOS()
+                                : AudioSamples(),
+      player_fixture.HasVideo() ? VideoSamples(0, kSamplesToWrite).WithEOS()
+                                : VideoSamples()));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
+}
+
+std::vector<SbPlayerTestConfig> GetSupportedTestConfigs() {
+  static std::vector<SbPlayerTestConfig> supported_configs;
+  if (supported_configs.size() > 0) {
+    return supported_configs;
+  }
+
+  std::vector<const char*> key_systems;
+  key_systems.push_back("");
+  key_systems.insert(key_systems.end(), kKeySystems,
+                     kKeySystems + SB_ARRAY_SIZE_INT(kKeySystems));
+
+  for (auto key_system : key_systems) {
+    std::vector<SbPlayerTestConfig> configs =
+        GetSupportedSbPlayerTestConfigs(key_system);
+    supported_configs.insert(supported_configs.end(), configs.begin(),
+                             configs.end());
   }
-  ASSERT_NO_FATAL_FAILURE(player_fixture_->WaitForPlayerEndOfStream());
+  return supported_configs;
 }
 
-std::string GetSbPlayerTestConfigName(
+std::string GetTestConfigName(
     ::testing::TestParamInfo<SbPlayerTestConfig> info) {
-  const char* audio_filename = std::get<0>(info.param);
-  const char* video_filename = std::get<1>(info.param);
-  SbPlayerOutputMode output_mode = std::get<2>(info.param);
+  const SbPlayerTestConfig& config = info.param;
+  const char* audio_filename = std::get<0>(config);
+  const char* video_filename = std::get<1>(config);
+  const SbPlayerOutputMode output_mode = std::get<2>(config);
+  const char* key_system = std::get<3>(config);
   std::string name(FormatString(
-      "audio_%s_video_%s_output_%s", audio_filename, video_filename,
+      "audio_%s_video_%s_output_%s_key_system_%s", audio_filename,
```

Comment:
Does this work if `audio_filename` is nullptr?  Can we consolidate this with the code at line 37, maybe add operator<< for SbPlayerTestConfig?

Reply:
As we already have formatted test name, it's not necessary to log out all configs.